sdk/storage/Azure.Storage.Blobs/tests/BlobUriBuilderTests.cs

```diff
@@ -186,6 +187,27 @@ public void BlobUriBuilder_RegularUrl_SasTest()
             Assert.AreEqual(originalUri, newUri);
         }
 
+        [Test]
+        public void BlobUriBuilder_RegularUrl_MalformedSubdomain()
+        {
+            // core and blob swapped
+            var blobUriBuilder1 = new BlobUriBuilder(new Uri("https://account.core.blob.windows.net/container/blob"));
+
+            // account and blob swapped
+            var blobUriBuilder2 = new BlobUriBuilder(new Uri("https://blob.account.core.windows.net/container/blob"));
+
+            // wrong service
+            var blobUriBuilder3 = new BlobUriBuilder(new Uri("https://account.file.core.windows.net/container/blob"));
+
+            // empty service
+            var blobUriBuilder4 = new BlobUriBuilder(new Uri("https://account./container/blob"));
```

Comment:
Is that a valid URI?

Reply:
Yes, at least the URI ctor doesn't throw.